app/src/main/java/dev/aurakai/auraframefx/ui/viewmodel/HomeScreenTransitionViewModel.kt

```
@@ -1,18 +1,56 @@
+
+// Extension function to get serializer for HomeScreenTransitionConfig
+@OptIn(kotlinx.serialization.ExperimentalSerializationApi::class)
+private fun HomeScreenTransitionConfig.toJson(): String {
+    return Json.encodeToString(this)
+}
+
+@OptIn(kotlinx.serialization.ExperimentalSerializationApi::class)
+private fun String.toHomeScreenTransitionConfig(): HomeScreenTransitionConfig? {
+    return try {
+        Json.decodeFromString<HomeScreenTransitionConfig>(this)
+    } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)